include/verilated_vpi.cpp

```diff
@@ -516,9 +516,38 @@ class VerilatedVpioModuleIter final : public VerilatedVpio {
     }
 };
 
+class VerilatedVpioScopeIter final : public VerilatedVpio {
+    const std::vector<const VerilatedScope*>* m_vec;
+    std::vector<const VerilatedScope*>::const_iterator m_it;
+
+public:
+    explicit VerilatedVpioScopeIter(const std::vector<const VerilatedScope*>& vec)
+        : m_vec{&vec} {
+        m_it = m_vec->begin();
+    }
+    ~VerilatedVpioScopeIter() override = default;
+    static VerilatedVpioScopeIter* castp(vpiHandle h) {
+        return dynamic_cast<VerilatedVpioScopeIter*>(reinterpret_cast<VerilatedVpio*>(h));
```

Comment:
Fix codacity warning: The class 'VerilatedVpioScopeIter' defines member function with name 'castp' also defined in its parent class 'VerilatedVpio'.

Reply:
Don't all of these classes do that?

Reply:
If so, fine.